core-bundle/src/Resources/contao/dca/tl_content.php

```diff
@@ -2040,4 +2034,15 @@ public function toggleIcon($row, $href, $label, $title, $icon, $attributes)
 
 		return '<a href="' . $this->addToUrl($href) . '" title="' . StringUtil::specialchars($title) . '" onclick="Backend.getScrollOffset();return AjaxRequest.toggleField(this,true)">' . Image::getHtml($icon, $label, 'data-icon="' . Image::getPath('visible.svg') . '" data-icon-disabled="' . Image::getPath('invisible.svg') . '" data-state="' . ($row['invisible'] ? 0 : 1) . '"') . '</a> ';
 	}
+
+	private function getAliasReferences(): array
+	{
+		if (null === self::$cteAliasCache)
+		{
+			$connection = System::getContainer()->get('database_connection');
+			self::$cteAliasCache = array_flip($connection->fetchFirstColumn("SELECT cteAlias FROM tl_content WHERE type='alias' AND (ptable='tl_article' OR ptable='') GROUP BY cteAlias"));
```

Comment:
I think I’d store the IDs as values, not keys. 
Or at least set all the values to `true` (e.g. using array_map).
Because having the row numbers as values seems counter intuitive to me.

But just a nitpick, feel free to ignore. ☺️

Reply:
Yes, key is fine, but the values should all be set to `true` then IMO. But maybe that’s just my OCD 😅

Reply:
Yes, they should be `true` then. At least that is how I do it, too.

Reply:
Probably yes.

Reply:
I could also use

```php
$connection->fetchAllKeyValue("SELECT cteAlias, TRUE FROM tl_content WHERE type='alias' AND (ptable='tl_article' OR ptable='') GROUP BY cteAlias");
```

then the array will be filled with `1`, i.e.

```php
[
    123 => 1,
    234 => 1,
    345 => 1,
]
```

Reply:
Yes, let‘s use what is already there. 👍